cmd/kube-apiserver/app/server.go

```diff
@@ -326,6 +326,7 @@ func CreateKubeAPIServerConfig(s *options.ServerRunOptions, nodeTunneler tunnele
 	}
 
 	var issuer serviceaccount.TokenGenerator
+	var apiAudiences []string
 	if s.ServiceAccountSigningKeyFile != "" || s.Authentication.ServiceAccounts.Issuer != "" {
```

Comment:
`|| len(s.Authentication.ServiceAccounts.APIAudiences) > 0`?

Reply:
`|| utilfeature.DefaultFeatureGate.Enabled(features.TokenRequest)`

Reply:
I did the first one. However I don't like requiring these flags to be set when the feature gate is enabled. That's going to break people using AllAlpha. Do we want to require these flags to be passed when the feature leaves alpha?

Reply:
Fair point, hadn't really considered the future lifecycle of the feature gate getting auto-enabled. If that's the case, disconnecting the "default to TLS private key as verifying key" based on whether they provided a signing key would be better … when they take positive action to use the new token signing system, they must explicitly provide the token verifying key